macros/macros_test.go

```diff
@@ -0,0 +1,72 @@
+package macros
+
+import (
+	"fmt"
+	"testing"
+	"text/template"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const validEndpointTemplate = "http://{{.Host}}/publisher/{{.PublisherID}}"
+const invalidEndpointTemplate = "http://{{Host}}/publisher/{{PublisherID}}"
+
+func TestBuildTemplate(t *testing.T) {
+	testCases := []struct {
+		templateStr string
+		hasError    bool
+	}{
+		{templateStr: validEndpointTemplate, hasError: false},
+		{templateStr: invalidEndpointTemplate, hasError: true},
+	}
+
+	for _, test := range testCases {
+		_, err := BuildTemplate(test.templateStr)
+
+		if test.hasError {
+			assert.NotNil(t, err, fmt.Sprintf("Building a template for %s should return an error", test.templateStr))
+		} else {
+			assert.Nil(t, err, fmt.Sprintf("Building a template for %s shouldn't return an error", test.templateStr))
+		}
+	}
+}
+
+func TestContainsMacro(t *testing.T) {
+	testCases := []struct {
+		str    string
+		result bool
+	}{
+		{str: validEndpointTemplate, result: true},
+		{str: invalidEndpointTemplate, result: false},
+	}
+
+	for _, test := range testCases {
+		assert.Equal(t, ContainsMacro(test.str), test.result, fmt.Sprintf("ContainsMacro should return %t for %s", test.result, test.str))
+	}
+}
+
+func TestResolveMacros(t *testing.T) {
+	endpointTemplate, _ := BuildTemplate(validEndpointTemplate)
+
+	testCases := []struct {
+		aTemplate template.Template
+		params    interface{}
+		result    string
+		hasError  bool
+	}{
+		{aTemplate: *endpointTemplate, params: EndpointTemplateParams{Host: "SomeHost", PublisherID: 1}, result: "http://SomeHost/publisher/1", hasError: false},
+		{aTemplate: *endpointTemplate, params: UserSyncTemplateParams{GDPR: "SomeGDPR", GDPRConsent: "SomeGDPRConsent"}, result: "", hasError: true},
+	}
+
+	for _, test := range testCases {
+		res, err := ResolveMacros(test.aTemplate, test.params)
+
+		if test.hasError {
+			assert.NotNil(t, err, "Error shouldn't be nil")
+			assert.Empty(t, res, "Result should be empty")
+		} else {
+			assert.Nil(t, err, "Err should be nil")
+			assert.Equal(t, res, test.result, fmt.Sprintf("String after resolving macros should be %s", test.result))
```

Comment:
no need to `fmt.Sprintf` here... the `assert.*` methods have this functionality already... so you can just:

```
assert.Equal(t, res, test.result, "String after resolving macros should be %s", test.result)
```

Reply:
Ah, good to know. Thanks!